carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/router/HttpDispatcher.java

```diff
@@ -21,26 +21,50 @@
 
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.SimpleChannelInboundHandler;
-import io.netty.handler.codec.http.HttpRequest;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.util.AttributeKey;
+
 
 /**
  * HttpDispatcher that invokes the appropriate http-handler method. The handler and the arguments are read
  * from the {@code RequestRouter} context.
  */
+public class HttpDispatcher extends SimpleChannelInboundHandler<HttpObject> {
 
-public class HttpDispatcher extends SimpleChannelInboundHandler<HttpRequest> {
+    private HttpMethodInfoBuilder httpMethodInfoBuilder;
 
-    private static final Logger LOG = LoggerFactory.getLogger(HttpDispatcher.class);
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws HandlerException {
+        Object httpMethodInfoBuilderObj = ctx.pipeline()
+                .context(RequestRouter.class)
+                .attr(AttributeKey.valueOf(RequestRouter.METHOD_INFO_BUILDER))
+                .get();
+        if (httpMethodInfoBuilderObj instanceof HttpMethodInfoBuilder) {
+            httpMethodInfoBuilder = (HttpMethodInfoBuilder) httpMethodInfoBuilderObj;
+            if (msg instanceof FullHttpRequest) {
+                FullHttpRequest fullHttpRequest = (FullHttpRequest) msg;
+                httpMethodInfoBuilder
+                        .httpRequest(fullHttpRequest)
+                        .build()
+                        .invoke();
+            } else if (msg instanceof HttpContent) {
+                httpMethodInfoBuilder
```

Comment:
I think, since it is method chaining it comes in separate lines when formatting..